hpx/lcos/detail/future_await_traits.hpp

```diff
@@ -13,7 +13,7 @@
 #include <hpx/traits/future_access.hpp>
 
 #if defined(HPX_HAVE_EMULATE_COROUTINE_SUPPORT_LIBRARY)
-#include <hpx/util/await_traits.hpp>
+#include <hpx/util_misc/await_util_misc.hpp>
```

Comment:
This should probably be `hpx/util_misc/await_traits.hpp`?

Reply:
Should we use sub-directories, i.e. `<hpx/util/misc/…>`?

Reply:
I'm torn on this one. The other utilities got their own top-level directories (i.e. under `hpx`), and not in `hpx/util`, so by that logic keeping `util_misc` would fit. Ideally this module would have a better name than `util_misc`...

Reply:
Perhaps, just `util` would do now?

Reply:
Yeah, I suppose that would be better. @aurianer would you mind...? I think we can merge this after that.